dataretrieval/nwis.py

```diff
@@ -447,21 +447,43 @@ def get_pmcodes(parameterCd='All', **kwargs):
     Parameters (Additional parameters, if supplied, will be used as query parameters).
     ----------
         parameterCd: string or listlike
+        parameterNm: string
     Returns:
         DataFrame containing the USGS parameter codes and Metadata as tuple
     """
-    payload = {'radio_pm_search' : 'pm_search',
-               'pm_group' : 'All+--+include+all+parameter+groups',
-               'pm_search' : parameterCd,
-               'casrn_search' : None,
-               'srsname_search' : None,
-               'show' :  ['parameter_group_nm', 'casrn', 'srsname','parameter_units', 'parameter_nm'],
-               'format' : 'rdb'}
+    url = 'https://help.waterdata.usgs.gov/code/parameter_cd_nm_query?'
     
-    payload.update(kwargs)
-    url = WATERDATA_URL + 'pmcodes/pmcodes'
+    if parameterCd is None and parameterNm is None:
+        raise TypeError('Query must specify a parameter code (parameterCd = ) or name (parameterNm = )')
+        
+    if parameterCd is not None and parameterNm is not None:
+        raise TypeError('Query must specify a parameter name or number, not both)')
+    
+    if parameterNm is None and parameterCd is not None: # querying based on a parameter code or list of codes
```

Comment:
You have already checked that only parameterCd or parameterNm is defined above.  No need to check check again.  Just check if parameterNm is defined.  It will be easier to read and follow.

Reply:
This makes perfect sense, I'll fix this, and the 'else' you mention below.